### PR TITLE
app-mobilephone/heimdall: updated links and RDEPEND

### PR DESCRIPTION
**Pull request by scardracs**
Bumping to heimdall-1.4.2-r1

Package-Manager: Portage-3.0.17, Repoman-3.0.2
Signed-off-by: Marco Scardovi <marco@scardovi.com>

### CODE REVIEW

**Comment by ScardracS** [comment]
@Polynomial-C I have made a new PR and restored the if/else

**Review by Polynomial-C — APPROVED** [review]

File: app-mobilephone/heimdall/metadata.xml [hunk 1]
```diff
@@ -1,8 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-	<!-- maintainer-needed -->
+	<maintainer type="person" proxied="yes">
+		<email>marco@scardovi.com</email>
+		<name>Marco Scardovi</name>
+	</maintainer>
+	<maintainer type="project">
```
Comment (jkroonza) [inline app-mobilephone/heimdall/metadata.xml:8]: proxied="proxy"

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		-DDISABLE_FRONTEND=$(usex !gui)
+	)
+	cmake_src_configure
+}
+
+src_install() {
+	dobin "${BUILD_DIR}"/bin/heimdall
+	use gui && dobin "${BUILD_DIR}"/bin/heimdall-frontend
+	insinto "$(get_udevdir)/rules.d"
+	doins heimdall/60-heimdall.rules
```
Comment (jkroonza) [inline app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild:47]: udev_dorules heimdall/60-heimdall.rules

And you get to drop the insinto as well.

**Review by jkroonza — CHANGES_REQUESTED** [review]
Almost there!

File: app-mobilephone/heimdall/metadata.xml (hunk 1 above)
Comment (ScardracS) [reply app-mobilephone/heimdall/metadata.xml:8]: Resolved

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild (hunk 2 above)
Comment (ScardracS) [reply app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild:47]: Resolved

**Review by jkroonza — CHANGES_REQUESTED** [review]
Squash the two commits:

git rebase -i master

Set the second commit to fixup (default is pick).  This will merge the second commit into the first, and keep the first one's commit message.

git push --force

This will update upstream, without --force git will reject the push since it's first removing the two commits you already pushed, and then adding a new one (effectively), so --force is required to tell git this is what you actually want.

**Comment by ScardracS** [comment]
> Squash the two commits:
> 
> git rebase -i master
> 
> Set the second commit to fixup (default is pick). This will merge the second commit into the first, and keep the first one's commit message.
> 
> git push --force
> 
> This will update upstream, without --force git will reject the push since it's first removing the two commits you already pushed, and then adding a new one (effectively), so --force is required to tell git this is what you actually want.

Thanks to remembering me! :)

File: app-mobilephone/heimdall/metadata.xml (hunk 1 above)
Comment (00-matt) [reply app-mobilephone/heimdall/metadata.xml:8]: I think `proxied="proxy"` goes on the proxy-maint maintainer element, and `proxied="yes"` goes on your own maintainer element.

**Review by jkroonza — COMMENTED** [review]
I'm happy, but I'm not in a position to approve.

File: app-mobilephone/heimdall/metadata.xml (hunk 1 above)
Comment (ScardracS) [reply app-mobilephone/heimdall/metadata.xml:8]: Oh, good point. I will take a look rn

**Comment by ScardracS** [comment]
@00-matt could you review the PR?

**Comment by 00-matt** [comment]
On Fri, 19 Mar 2021, at 13:05, ScardracS wrote:
> @00-matt <https://github.com/00-matt> could you review the PR?

It looks good to me, but I'm not a member of
proxy-maintainers (or even a developer at all) so I'm not sure if
my approval or review is worth anything :)

**Comment by ScardracS** [comment]
> On Fri, 19 Mar 2021, at 13:05, ScardracS wrote:
> > @00-matt <https://github.com/00-matt> could you review the PR?
> 
> It looks good to me, but I'm not a member of
> proxy-maintainers (or even a developer at all) so I'm not sure if
> my approval or review is worth anything :)
> 

Ops, I thought you actually were a part of it 😅

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	SRC_URI="https://github.com/Benjamin-Dobell/Heimdall/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/Heimdall-${PV}"
+else
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Benjamin-Dobell/Heimdall.git"
+fi
+
+DESCRIPTION="Tool suite used to flash firmware onto Samsung devices"
+HOMEPAGE="https://glassechidna.com.au/heimdall/"
```
Comment (juippis) [inline app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild:18]: Please also add the Github page as 2ndary HOMEPAGE here.

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="gui"
+
+RDEPEND="
+	virtual/libusb:1=
+	sys-libs/zlib
```
Comment (juippis) [inline app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild:26]:
```suggestion
	sys-libs/zlib
	virtual/libusb:1=
```

**Review by juippis — COMMENTED** [review]
Comments apply to -9999 ebuild too, looks good otherwise!

**Comment by juippis** [comment]
Also I'd split this into 3 commits. 
1: revbump 1.4.2,
2: sync -9999 ebuild, 
3: modify metadata.xml.

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild (hunk 3 above)
Comment (scardracs) [reply app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild]: Added for both 1.4.2-r1 and -9999

File: app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild (hunk 4 above)
Comment (scardracs) [reply app-mobilephone/heimdall/heimdall-1.4.2-r1.ebuild]: Fixed on both

**Comment by scardracs** [comment]
> Also I'd split this into 3 commits.
> 1: revbump 1.4.2,
> 2: sync -9999 ebuild,
> 3: modify metadata.xml.

Now I have 3 differents commits, but the first one still show the adding of 1.4.2-r1... Is this what you mean or I have to drop first commit and redo it with just changing metadata.xml?

**Review by juippis — APPROVED** [review]
Yeah no the split didn't quite work, I'll squash them into one.

**Comment by juippis** [comment]
85faca572fced4fcf82dc3964ac4461f4700b88a

**Pull request closed without merging**